resources/shared/workload-testing-utils.mjs

```diff
@@ -0,0 +1,166 @@
+import { TestRunner } from "./test-runner.mjs";
+import { Params } from "./params.mjs";
+
+/**
+ * BenchmarkTestStep
+ *
+ * A single test step, with a common interface to interact with.
+ */
+export class BenchmarkTestStep {
+    constructor(name, run) {
+        this.name = name;
+        this.run = run;
+    }
+
+    async runAndRecord(params, suite, test, callback) {
+        const testRunner = new TestRunner(null, null, params, suite, test, callback);
+        const result = await testRunner.runTest();
+        return result;
+    }
+}
+
+/**
+ * BenchmarkTestSuite
+ *
+ * A single test suite that contains one or more test steps.
+ */
+export class BenchmarkTestSuite {
+    constructor(name, tests) {
+        this.name = name;
+        this.tests = tests;
+    }
+
+    record(_test, syncTime, asyncTime) {
+        const total = syncTime + asyncTime;
+        const results = {
+            tests: { Sync: syncTime, Async: asyncTime },
+            total: total,
+        };
+
+        return results;
+    }
+
+    async runAndRecord(params, onProgress) {
+        const measuredValues = {
+            tests: {},
+            total: 0,
+        };
+        const suiteStartLabel = `suite-${this.name}-start`;
+        const suiteEndLabel = `suite-${this.name}-end`;
+
+        performance.mark(suiteStartLabel);
+
+        for (const test of this.tests) {
+            const result = await test.runAndRecord(params, this, test, this.record);
+            measuredValues.tests[test.name] = result;
+            measuredValues.total += result.total;
+            onProgress?.(test.name);
+        }
+
+        performance.mark(suiteEndLabel);
+        performance.measure(`suite-${this.name}`, suiteStartLabel, suiteEndLabel);
+
+        return {
+            type: "suite-tests-complete",
+            status: "success",
+            result: measuredValues,
+            suitename: this.name,
+        };
+    }
+}
+
+/**
+ * BenchmarkSuitesManager
+ *
+ * A collection of test suites for a single workload.
+ */
+export class BenchmarkSuitesManager {
+    constructor(name, suites) {
+        this.name = name;
+        this.suites = suites;
+    }
+
+    getSuiteByName(name) {
+        return this.suites.find((suite) => suite.name === name);
+    }
+}
+
+/**
+ * Helper Methods
```

Comment:
I'd rather have a separate file for these helper methods and rename this file to "benchmark.mjs" or something since all classes here will have Benchmark prefix then.

Reply:
done